packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -377,15 +392,18 @@ contract Deploy is Deployer {
             _portal: OptimismPortal(payable(portal))
         });
 
-        require(address(messenger.PORTAL()) == portal);
-        require(address(messenger.portal()) == portal);
-
-        bytes32 xdmSenderSlot = vm.load(address(messenger), bytes32(uint256(204)));
-        require(address(uint160(uint256(xdmSenderSlot))) == Constants.DEFAULT_L2_SENDER);
-
         save("L1CrossDomainMessenger", address(messenger));
         console.log("L1CrossDomainMessenger deployed at %s", address(messenger));
 
+        // Override the `L1CrossDomainMessenger` contract to the deployed implementation. This is necessary
+        // to check the `L1CrossDomainMessenger` implementation alongside dependent contracts, which
+        // are always proxies.
+        Types.ContractSet memory contracts = _proxiesUnstrict();
+        contracts.L1CrossDomainMessenger = address(messenger);
```

Comment:
I would argue that needing to do this means that we need to rethink our abstractions here